be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -175,8 +218,114 @@ Status LakePersistentIndex::try_replace(size_t n, const Slice* keys, const Index
     return Status::OK();
 }
 
-Status LakePersistentIndex::major_compact(int64_t min_retain_version) {
+std::unique_ptr<sstable::Iterator> LakePersistentIndex::prepare_merging_iterator() {
+    sstable::ReadOptions read_options;
+    std::vector<sstable::Iterator*> iters;
+    auto max_compaction_versions = config::lake_pk_index_sst_max_compaction_versions;
+    iters.reserve(max_compaction_versions);
+    for (const auto& sstable : _sstables) {
+        sstable::Iterator* iter = sstable->new_iterator(read_options);
+        iters.emplace_back(iter);
+        if (iters.size() >= max_compaction_versions) {
+            break;
+        }
+    }
+    sstable::Options options;
+    sstable::Iterator* iter = sstable::NewMergingIterator(options.comparator, &iters[0], iters.size());
+    std::unique_ptr<sstable::Iterator> iter_ptr = nullptr;
+    iter_ptr.reset(iter);
```

Comment:
```suggestion
    std::unique_ptr<sstable::Iterator> iter_ptr(sstable::NewMergingIterator(options.comparator, &iters[0], iters.size()));
```

RAAI is perfectly for the case, avoid raw unmanaged pointer in between.